pkgs/development/compilers/vlang/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, glfw, freetype, curl }:
+
+let
+    # V compiler source translated to C for bootstrap.
+    vc = fetchFromGitHub {
+        owner = "vlang";
+        repo = "vc";
+        rev = "a0b6d1ac1c43380134c4c4f584b5abde8b438d1e";
+        sha256 = "0i5g38lfqbp8mzn8xas3sghmd6rdb2s8kr9ih25b0wxcjydmfv6h";
+    };
+in stdenv.mkDerivation rec {
+    pname = "vlang";
+    version = "0.1.4";
+
+    src = fetchFromGitHub {
+        owner = "vlang";
+        repo = "v";
+        rev = "v${version}";
+        sha256 = "0sip76rxpx5w7vy9s9vpib0ibfr5r2msfl81wcd966hg7p6cd5i6";
+    };
+
+    buildInputs = [ glfw freetype curl ];
+
+    buildPhase = ''
+        export HOME=`pwd`
+        cd compiler
+        cc -std=gnu11 -w -o v ${vc}/v.c
```

Comment:
This is not easily overridable. Consider replacing `${vc}` with `$vc` and moving `vc` to derivation attribute.